holidays/countries/finland.py

```diff
@@ -97,6 +102,83 @@ def _populate_public_holidays(self):
         # Second Day of Christmas.
         self._add_christmas_day_two(tr("Tapaninpäivä"))
 
+    def _populate_unofficial_holidays(self):
+        if self._year >= 1854:
+            # Runeberg Day
+            self._add_holiday_feb_5(tr("Runebergin päivä"))
+
+        if self._year >= 1860:
+            # Kalevala Day, Day of Finnish Culture
+            self._add_holiday_feb_28(tr("Kalevalan päivä, suomalaisen kulttuurin päivä"))
+
+        if self._year >= 2007:
+            #  Minna Canth Day, Day of Equality
+            self._add_holiday_mar_19(tr("Minna Canthin päivä, tasa-arvon päivä"))
+
+        if self._year >= 1978:
+            # Mikael Agricola Day, Day of the Finnish Language
+            self._add_holiday_apr_9(tr("Mikael Agricolan päivä, suomen kielen päivä"))
+
+        if self._year >= 1987:
+            # National War Veterans' Day
+            self._add_holiday_apr_27(tr("Kansallinen veteraanipäivä"))
+
+        if self._year >= 2019:
+            # Europe Day
+            self._add_holiday_may_9(tr("Eurooppa-päivä"))
+
+        if self._year >= 1918:
+            # Mothers' Day
+            self._add_holiday_may_12(tr("Äitienpäivä"))
+
+        if self._year >= 1952:
+            # J. V. Snellman Day, Day of Finnish Heritage
+            self._add_holiday_may_12(tr("J.V. Snellmanin päivä, suomalaisuuden päivä"))
+
+        if self._year >= 1977:
+            # Remembrance Day
+            self._add_holiday_3rd_sun_of_may(tr("Kaatuneitten muistopäivä"))
+
+        if self._year >= 1942:
+            # Flag Day of the Finnish Defence Forces
+            self._add_holiday_jun_6(tr("Puolustusvoimain lippujuhlan päivä"))
+
+        if self._year >= 1992:
+            # Eino Leino Day, Day of Summer and Poetry
+            self._add_holiday_jul_6(tr("Eino Leinon päivä, runon ja suven päivä"))
+
+        if self._year >= 2013:
+            # Finland's Nature Day
+            self._add_holiday_last_sat_of_aug(tr("Suomen luonnon päivä"))
+
+        if self._year >= 2016:
+            # Miina Sillanpää Day, Day of Civic Participation
+            self._add_holiday_oct_1(tr("Miina Sillanpään ja kansalaisvaikuttamisen päivä"))
+
+        if self._year >= 1950:
+            # Aleksis Kivi Day, Day of Finnish Literature
+            self._add_holiday_oct_10(tr("Aleksis Kiven päivä, suomalaisen kirjallisuuden päivä"))
+
+        if self._year >= 1987:
+            # United Nations Day
+            self._add_united_nations_day(tr("YK:n päivä"))
+
+        if self._year >= 1908:
+            # Finnish Swedish Heritage Day, svenska dagen
+            self._add_holiday_nov_6(tr("Ruotsalaisuuden päivä, Kustaa Aadolfin päivä"))
+
+        if self._year >= 1949:
+            # Fathers' Day
+            self._add_holiday_nov_11(tr("Isänpäivä"))
```

Comment:
```suggestion
            self._add_holiday_2nd_sun_of_nov(tr("Isänpäivä"))
```
Second Sunday in November.